rlog/state-separation.mdx

```diff
@@ -0,0 +1,1255 @@
+transactions or sensitive data management within smart contracts.
+
+In a private execution, the user initiates a transaction involving private data. The transaction spends existing UTXOs and creates new ones, all represented as 
+leaves in the Merkle tree. The zkVM generates a zero-knowledge proof to validate the transaction without revealing private data. This proof is submitted to the sequencer, 
+which verifies the proof to ensure the transaction's validity. Upon successful verification, the nullifier is added to the nullifier set, and the private state is updated 
+with the new Merkle root.
+
+#### 3. Shielded Executions
+
+Shielded executions create a layer of privacy for the outputs by allowing interactions between the public and private states. When a transaction occurs in a shielded execution, 
+details of the transaction are processed within the private state, ensuring that sensitive information remains confidential. Only the necessary details are shared with the public state, 
+often in a masked or encrypted form. This approach allows for the validation of the transaction without revealing critical data, thus preserving the privacy of the involved parties.
+
+The workflow for shielded executions begins with the user initiating a transaction that reads from the public state and prepares to write to the private state. Public data is accessed, 
+and the private state is prepared to receive new data. The zkVM generates a zero-knowledge proof to hide the receiver’s identity. This proof is submitted to the sequencer, which verifies 
+the proof to ensure the transaction's validity. If valid, the private state is updated with the new data, while the public state reflects the change without revealing private details. 
+This type of execution is particularly useful for scenarios where the receiver’s identity needs to be hidden, such as in anonymous donation systems or confidential data storage.
+
+#### 4. Deshielded Executions
+
+Deshielded executions operate in the opposite manner of shielded executions, where data is read from the private state and written to the public state. This execution type is useful 
+in situations where the sender's identity needs to be kept confidential while making the transaction results publicly visible.
+
+In a deshielded execution, the user initiates a transaction that reads from the private state and prepares to write to the public state. Private data is accessed, 
+and the transaction details are prepared. The zkVM generates a zero-knowledge proof to hide the sender’s identity. This proof is then submitted to the sequencer, 
+which verifies the proof to ensure the transaction's validity. Once verified, the public state is updated with the new data, reflecting the change while keeping the sender’s 
+details confidential. This can be useful when transparency is needed, such as when auditing or proving certain aspects of a transaction to a wider audience. 
```

Comment:
Just a thought and i might not understand it fully, but would voting be a good use for this one? You want the record to show that there was a vote cast but you don't want the world to know it was you who voted for something - secret ballot basically.